aten/src/ATen/native/Loss.cpp

```diff
@@ -70,6 +74,76 @@ TORCH_IMPL_FUNC(smooth_l1_loss_out)
   }
 }
 
+Tensor& margin_ranking_loss_out
+(const Tensor& input1,
+ const Tensor& input2,
+ const Tensor& target,
+ double margin,
+ int64_t reduction,
+ Tensor& result
+) {
+  TORCH_CHECK(
+    !input1.is_complex() &&
+    !input2.is_complex() &&
+    !target.is_complex(),
+    "margin_ranking_loss expects inputs to be non-complex");
+
+  // Disallow broadcasting:
+  // https://github.com/pytorch/pytorch/pull/64975
+  TORCH_CHECK(
+    input1.sizes() == input2.sizes() &&
+    input1.sizes() == target.sizes(),
+    "margin_ranking_loss: all input tensors should have same size but got sizes"
+    ": input1: ", input1.sizes(),
+    ", input2: ", input2.sizes(),
+    ", target: ", target.sizes());
+
+  const auto common_type1 = promoteTypes(input1.scalar_type(), target.scalar_type());
+  const auto common_type2 = promoteTypes(common_type1, input2.scalar_type());
```

Comment:
yeah, it's hard for me to judge the impact here, so i'm mostly relying on existing tests and the feedback that others provide here, which is why these names are used too -- to avoid making any incorrect assumptions

Reply:
if you see a problem with this or have a better idea for names, that's welcome

Reply:
I think it does, as `promoteTypes` is associative (because it is, right? I was thinking about this for some time and I managed to convince myself that it is, but I might be wrong).